File: HelpClient/src/app/core/models/Vehiculo.ts
import { Entity } from './common/Entity';

import { Componente } from './Componente';

export class Vehiculo extends Entity {


  code: string;
  nombre: string;
  componente: Componente;

  public setCode(code: string): void {
    this.code = code;
  }

  public getCode(): string {
    return this.code;
  }

  public setNombre(nombre: string): void {
    this.nombre = nombre;
  }

  public getNombre(): string {
    return this.nombre;
  }

  public setComponente(componente: Componente): void {
    this.componente = componente;
  }

  public getComponente(): Componente {
    return this.componente;
  }


  static fill(v: any): Vehiculo {
    let vehiculo = new Vehiculo();
    vehiculo.setNombre(v.nombre);
    return vehiculo;
  }

}

File: HelpServer/src/main/java/es/malaga/diputacion/ae/help/modelos/entity/Vehiculo.java
package es.malaga.diputacion.ae.help.modelos.entity;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "VEHICULOS", schema = "MASTER")
public class Vehiculo implements Serializable {

	@Id
	//@GeneratedValue(strategy = GenerationType.SEQUENCE)
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SEQ_VEHICULOS")
	private Long id;

	@Column(name = "CODE")
	private String code;

	@Column(name = "NOMBRE")
	private String nombre;

	@Column(name = "FECHA_REGISTRO")
	private Date fechaRegistro;

	@ManyToOne
	@JoinColumn(name = "ID_COMPONENTE")
	private Componente componente;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getCode() {
		return code;
	}

	public void setCode(String code) {
		this.code = code;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public Date getFechaRegistro() {
		return fechaRegistro;
	}

	public void setFechaRegistro(Date fechaRegistro) {
		this.fechaRegistro = fechaRegistro;
	}

	public Componente getComponente() {
		return componente;
	}

	public void setComponente(Componente componente) {
		this.componente = componente;
	}

	private static final long serialVersionUID = 1L;

}
